### ISSUE TITLE
cblas_ddot different results

### ISSUE THREAD

**Issue by miro-janosik** [xianyi/OpenBLAS#2953]
Good afternoon

my problem shares the error with https://github.com/xianyi/OpenBLAS/issues/1168 - I also get error "Cholesky decomposition failed. Maybe matrix is not positive definite" when loading matrix with Kaldi.

My problem is that I have code compiled for Windows 10 (where it works), for CentOS 7 (where it works too), and after compiling it for Ubuntu 20 machine there it fails with this error, while loading the same matrix data. From what I see the operation call cblas_Xdot() returns different result for the same data .

What would be a best way to track down the problem please? I am trying to use the same version on all platforms (it is xianyi-OpenBLAS-eebc189-0-3-5 ) and I tried to build it with just a 'make' or 'make USE_THREAD=1 DYNAMIC_ARCH=1'.

When I run the tests I can see that cblas_ddot passed ... though, also when loading the matrices in Kaldi it works in first 30 matrices and fails on 31st.

I have a sample text output from the Cholesky decomposition - one run on windows, another on the linux, where it is clearly visible (in a diff tool like Meld or WinDiff) where results start to differ and Cholesky fails. It can be seen at http://bin.mypage.sk/FILES/localCholesky.zip

  cheers and thanks for hints, Miro

**Comment by martin-frbg** [comment]
Is there a reason why you need to use version 0.3.5 from almost 2 years ago ? In particular, 0.3.6 had a fix for sloppy declarations of register use in the dot microkernel (and several others) on x86_64 (where the consequences of the sloppyness would depend on the optimisations performed by the compiler - the newer the more aggressive) and there have been a few fixes for race conditions in the multithreading.

**Comment by miro-janosik** [comment]
Hi, reason was that we jumped off from the version that was available in some of linux distributions at the time, and wanted to have the same versions on all other platforms too as it worked well.

Also Kaldi suggested using 0.3.5 until end of 2019 ( https://github.com/kaldi-asr/kaldi/pull/3642/files ), now it suggests 0.3.7.

I have now tried to use latest OpenBLAS 0.3.12 and problem is gone, this is a good enough solution for now for me.
